python/love/producer/love_producer_csc.py

```diff
@@ -0,0 +1,552 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["LoveProducerCSC"]
+
+import copy
+import asyncio
+import logging
+import datetime
+
+from typing import Any, Awaitable, Optional, Tuple
+
+from lsst.ts.salobj import Domain, Remote
+
+from . import LoveProducerBase
+from .producer_utils import get_data_type
+
+
+class LoveProducerCSC(LoveProducerBase):
+    """Specialized LOVE producer to deal with generic CSC behavior."""
+
+    def __init__(
+        self, domain: Domain, csc: str, log: Optional[logging.Logger] = None, **kwargs
+    ) -> None:
+        super().__init__(component_name=csc, log=log)
+
+        self.add_metadata(**kwargs)
+
+        self.remote: Remote = Remote(
+            domain,
+            csc,
+            index=kwargs.get("salindex", 0),
+            readonly=kwargs.get("remote_readyonly", True),
+        )
+
+        self._events_special_cases = {"evt_heartbeat"}
+
+        self._non_topic_data_stream = {}
+
+        self._revcode_topic_attribute_name_map: dict = dict()
+        self._template_manager_message: dict = dict()
+
+        self._need_reply_category = {"initial_state"}
+
+        self.periodic_data: list = self.get_periodic_data(**kwargs)
+        self.asynchronous_data: list = self.get_asynchronous_data(**kwargs)
+
+        self.store_last_sample_timeout: float = 0.5
+        self.heartbeat_timeout = 2.0
+        self.heartbeat_max_lost = 5
+
+        self.set_topic_name_revcode_mapping()
+        self.set_topic_template_manager_message_format()
+
+        self._heartbeat_monitor_task = None
+        self.done_task: asyncio.Future = asyncio.Future()
+
+        self.start_task: Awaitable = asyncio.create_task(self.start())
+
+    def should_reply_to_message_data(self, message_data: dict) -> bool:
+        """Determines whether a reply to message data should be sent.
+
+        Override base class default behavior.
+
+        Parameters
+        ----------
+        message_data: `dict`
+            Input dictionary with information used by the producer to determine
+            some action to take.
+
+        Returns
+        -------
+        `bool`
+            Should reply to message data?
+        """
+        category = message_data.get("category", None)
+        data = message_data.get("data", [dict()])[0]
+        csc = data.get("csc", None)
+        salindex = data.get("salindex", 0)
+
+        self.log.debug(
+            "Should reply to message_data: "
+            f"{category in self._need_reply_category}, "
+            f"{csc in self.reply_names}, "
```

Comment:
Why not use only `self.remote.salinfo.name` instead of self.reply_names?

Reply:
This is mainly to keep it generic. For instance, the ScriptQueue producer reply to more than just `self.remote.salinfo.name`, as it receives requests for ScriptQueue, ScriptQueueState and Script. 

https://github.com/lsst-ts/LOVE-producer/blob/a963f764d01aa3e87159d1860ff99007446a7ce4/python/love/producer/love_producer_script_queue.py#L740

Reply:
I see, thanks for the explanation! I'll resolve the conversation